File: projects/order-creation-queries/core/src/main/java/com/robinfood/core/mappers/OrderDetailChangedPortionMappers.java
package com.robinfood.core.mappers;

import com.robinfood.core.dtos.OrderDetailChangedPortionDTO;
import com.robinfood.core.entities.OrderDetailChangedPortionEntity;

import java.util.Objects;

public final class OrderDetailChangedPortionMappers {

    private OrderDetailChangedPortionMappers() {
        throw new IllegalStateException("Utility class");
    }

    public static OrderDetailChangedPortionDTO toOrderDetailChangedPortionDTO(
            OrderDetailChangedPortionEntity orderDetailChangedPortionEntity
    ) {
        if (Objects.isNull(orderDetailChangedPortionEntity)) {
            return null;
        }

        return new OrderDetailChangedPortionDTO(
                orderDetailChangedPortionEntity.getId(),
                orderDetailChangedPortionEntity.getName(),
                orderDetailChangedPortionEntity.getProductId(),
                orderDetailChangedPortionEntity.getSku(),
                orderDetailChangedPortionEntity.getUnitId(),
                orderDetailChangedPortionEntity.getUnitNumber()
        );
    }
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/mappers/OrderDetailChangedPortionMappers.kt
package com.robinfood.core.mappers

import com.robinfood.core.constants.GlobalConstants.DEFAULT_DOUBLE_VALUE
import com.robinfood.core.constants.GlobalConstants.DEFAULT_LONG_VALUE
import com.robinfood.core.dtos.OrderDetailChangedPortionDTO
import com.robinfood.core.entities.OrderDetailChangedPortionEntity

fun OrderDetailChangedPortionEntity.toOrderDetailChangedPortionDTO(): OrderDetailChangedPortionDTO? {
    return if (id == null) {
        null
    } else {
        return OrderDetailChangedPortionDTO(
                id = id,
                name = name.orEmpty(),
                parentId = parentId ?: DEFAULT_LONG_VALUE,
                sku = sku.orEmpty(),
                unitId = unitId ?: DEFAULT_LONG_VALUE,
                unitNumber = unitNumber ?: DEFAULT_DOUBLE_VALUE
        )
    }
}